### ISSUE TITLE
Creating a Gift Voucher with a custom amount results in a WSOD/GSOD

### ISSUE THREAD

**Issue by robzor** [verbb/gift-voucher#112]
### Describe the bug

I am just deploying my gift voucher code to our staging environment. I went to create the default custom gift voucher we are using, entered the name (we are not using custom fields) and then toggled 'Custom Amount' on. It then gave me the grey screen of death.

Here is the error:
```
[error] TypeError: Cannot assign string to property verbb\giftvoucher\elements\Voucher::$price of type ?float at /var/www/html/vendor/verbb/gift-voucher/src/helpers/VoucherHelper.php:55
Stack trace:
#0 /var/www/html/vendor/verbb/gift-voucher/src/controllers/VouchersController.php:190
```

Here is some of the info sent in the post request from the server logs:
```
'action' => 'gift-voucher/vouchers/save'
'typeId' => '1'
'redirect' => 'e8cac8b3dc7026a3578e2ac0eefd9e2812349bb133bff691cc00bbda05992192gift-voucher/vouchers'
'title' => 'Gift Voucher'
'slug' => 'gift-voucher'
'postDate' => [
'date' => ''
'timezone' => 'Europe/London'
'time' => ''
]
'expiryDate' => [
'date' => ''
'timezone' => 'Europe/London'
'time' => ''
]
'enabled' => '1'
'promotable' => '1'
'availableForPurchase' => '1'
'taxCategoryId' => '1'
'shippingCategoryId' => '1'
'sku' => ''
'customAmount' => '1'
'price' => ''
]
```

I managed to make it work by setting a arbitrary value of 1 on the voucher, then reopening the voucher and toggling the option to enabled.

### Steps to reproduce

1. Go to Gift Vouchers > Vouchers
2. Click 'New Voucher'
3. Enter details
4. Toggle Custom Amount on
5. Click Save

### Craft CMS version

4.3.6.1

### Plugin version

3.0.6

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/gift-voucher` requirement in `composer.json` to:

```
"require": {
  "verbb/gift-voucher": "dev-craft-4 as 3.0.6",
  "...": "..."
}
```

Then run `composer update`.

**Comment by robzor** [comment]
Thanks!

**Comment by robzor** [comment]
Hi there, I've finally just deployed the Gift Voucher feature to our production site and I had the same error when trying to create our custom gift voucher for the first time, and we are on version 3.0.6. I did the workaround of making a voucher with a value of 10 then changing it after but I thought I'd let you know.

`[error] TypeError: Cannot assign string to property verbb\giftvoucher\elements\Voucher::$price of type ?float at /var/www/html/vendor/verbb/gift-voucher/src/helpers/VoucherHelper.php:55`

looks like the line number is different from last time?
